src/main/java/school/hei/haapi/model/validator/UserValidator.java

```diff
@@ -0,0 +1,31 @@
+package school.hei.haapi.model.validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.model.User;
+import school.hei.haapi.model.exception.BadRequestException;
+
+@Component
+@AllArgsConstructor
+public class UserValidator {
+  private final Validator validator;
+
+  private void validateSomeUser(User user) {
+    List<ConstraintViolation<User>> violations = new ArrayList<>(validator.validate(user));
+    if (violations.size() == 1) {
+      throw new BadRequestException(violations.get(0).getMessage());
+    } else if (violations.size() > 1) {
+      throw new BadRequestException("Some fields are still empty");
```

Comment:
No. Create a message that contains all validation errors.

Reply:
Actually, the order of the constraint validation is random. For our test in TeacherIT, sometimes the first name field goes first and sometimes the last name field goes first. In prod, I don't think it matters but for test, it does because, the CI could failed !

Reply:
You can get the error message in your test. Then 'assertTrue(errorMessage.contains(badFieldName1)); assertTrue(errorMessage.contains(badFieldName2); ...'